### DISCUSSION TITLE
How to return one of a set of valid cultures?

### DISCUSSION THREAD

**Discussion by Kralizek** [AutoFixture/AutoFixture#1420] [Q&A] [answered]
I need to get a valid `CultureInfo` object.

I was thinking about using the `ElementsBuilder` to return one of the specified instances but this test fails

```csharp
[Test]
public void Test()
{
    var fixture = new Fixture();

    var validCultures = new[]
    {
        CultureInfo.GetCultureInfo("en-us"),
        CultureInfo.GetCultureInfo("en-es"),
        CultureInfo.GetCultureInfo("se-sv"),
        CultureInfo.GetCultureInfo("it-it"),
    };

    fixture.Customizations.Add(new ElementsBuilder<CultureInfo>(validCultures));

    var ci = fixture.Create<CultureInfo>();
    
    Assert.That(validCultures.Select(c => c.Name), Contains.Item(ci.Name));
}
```

Did I misundertand the purpose of `ElementsBuilder` or is it a bug? 

Years ago I used it to handle a similar scenario (see [here](https://github.com/Nybus-project/Nybus/blob/d3a8ba20492657e76b1413f3684d67ccd2e823cf/tests/Tests.Nybus.Engine.RabbitMq/CustomAutoMoqDataAttribute.cs#L35C22-L35C22))

**Comment by aivascu** [comment, marked as answer]
@Kralizek the request never reaches the `ElementsBuilder` since `Add` appends it to the end of the collection.
Use `.Insert(0, new ElementsBuilder...`

**Reply by Kralizek** [reply]
Oh, right 👍
